### PR TITLE
Feature/drawer

### PR DESCRIPTION
**Pull request by THMKAE**
ℹ️ Includes a (base) setup for a text & heading component, specific tag styling can be updated on a future release when needed

### CODE REVIEW

File: src/components/Text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
```
Comment (RobbeB) [inline src/components/Text/index.tsx]: I don't like putting both of these components in the index file

File: src/components/Text/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+type HeadingAsTagProps = {
+  asTag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+};
+
+type TextProps = {
+  size?: 'xxl' | 'xl' | 'l' | 'm' | 's' | 'xs' | 'xxs';
+} & HTMLAttributes<HTMLHeadingElement>;
+
+const StyledHeading = styled.h1<TextProps>`
+  font-weight: 600;
```
Comment (RobbeB) [inline src/components/Text/index.tsx]: Do they all have this font-weight?

File: src/components/Text/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    | 'address'
+    | 'pre';
+};
+
+const StyledText = styled.p<TextAsTagProps & TextProps>`
+  font-weight: 400;
+  font-family: ${props => props.theme.typography.fontFamily};
+  font-size: ${props => props.theme.text.fontSize.text[props.size || 'm']};
+  line-height: 1.6;
+  margin: ${props => (props.asTag === 'p' ? '0 0 16px 0' : '0')};
```
Comment (RobbeB) [inline src/components/Text/index.tsx]: this is going to need some work for the other elements

File: src/components/Text/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    | 'del'
+    | 'ins'
+    | 'blockquote'
+    | 'q'
+    | 'code'
+    | 'address'
+    | 'pre';
+};
+
+const StyledText = styled.p<TextAsTagProps & TextProps>`
```
Comment (RobbeB) [inline src/components/Text/index.tsx]: `TextProps` here refers to HTMLAttributes<**HTMLHeadingElement**>

File: src/components/Text/index.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { HTMLAttributes } from 'react';
+
+type HeadingAsTagProps = {
+  asTag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+};
+
+type TextProps = {
+  size?: 'xxl' | 'xl' | 'l' | 'm' | 's' | 'xs' | 'xxs';
```
Comment (RobbeB) [inline src/components/Text/index.tsx]: Maybe it's useful to also have the option to make the color secondary? (this is something we do quite often).
If you don't like it, atleast we should pass the classname so we can modify it with a styled component

File: src/components/Text/index.tsx (hunk 1 above)
Comment (THMKAE) [reply src/components/Text/index.tsx]: I'll put them into different files

File: src/components/Text/index.tsx (hunk 2 above)
Comment (THMKAE) [reply src/components/Text/index.tsx]: For now they do, but as mentioned, we can update this in the future.

File: src/components/Text/index.tsx (hunk 3 above)
Comment (THMKAE) [reply src/components/Text/index.tsx]: As mentioned, we can update this in the future.

File: src/components/Text/index.tsx (hunk 4 above)
Comment (THMKAE) [reply src/components/Text/index.tsx]: Good call, I'll update it

File: src/components/Text/index.tsx (hunk 5 above)
Comment (THMKAE) [reply src/components/Text/index.tsx]: The classname is already passed, I'll see what I can do about the colors.

**Pull request merged**